src/components/ScholarshipDetailCard.jsx

```diff
@@ -111,7 +112,16 @@ export default function ScholarshipDetailCard({ scholarship }) {
         <Button
           variant="outlined"
           className={classes.shareBtn}
-          startIcon={<Share />}>
+          startIcon={<Share />}
+          onClick={() =>
+            history.replace({
+              state: {
+                showShareDialog: true,
+                shareTitle: `${amount?.toString()} - ${name}`,
+                shareURL: `https://${window.location.hostname}/scholarships/${scholarship.id}`,
```

Comment:
You could also use a state variable like this:
```js
shareDialog: {
  title: ...,
  url: ...
}
```

and then simply check for the presence of `shareDialog`.

Reply:
I had made the changes but then kept the original version to tackle the glitch when closing the dialog.